### ISSUE TITLE
Out-of-date ChannelIndex docstring

### ISSUE THREAD

**Issue by mgr0dzicki** [NeuralEnsemble/python-neo#342]
There's an out-of-date description of attributes/properties in ChannelIndex docstring:

```
    *Required attributes/properties*:
        :channel_indexes: (numpy.array 1D dtype='i')
            Index of each channel in the attached signals.
    *Recommended attributes/properties*:
        :name: (str) A label for the dataset.
        :description: (str) Text description.
        :file_origin: (str) Filesystem path or URL of the original data file.
        :channel_names: (numpy.array 1D dtype='S')
            Names for each recording channel.
        :coordinates: (quantity array 2D (x, y, z))
            Physical or logical coordinates of all channels.
```

**Comment by JuliaSprenger** [comment]
You are right. Also the example usage cases are not up to date, at least this line is not working any more:
`chx = ChannelIndex(name='octotrode A')`

**Comment by apdavison** [comment]
Fixed in #383
